xmtp_mls/src/identity.rs

```diff
@@ -572,26 +574,27 @@ impl Identity {
     pub(crate) async fn rotate_and_upload_key_package<ApiClient: XmtpApi>(
         &self,
         provider: impl MlsProviderExt + Copy,
+        force_rotate: bool,
         api_client: &ApiClientWrapper<ApiClient>,
     ) -> Result<(), IdentityError> {
         let conn = provider.db();
+        if !conn.last_key_package_needs_rotation()? && !force_rotate {
+            // If not ready, mark it for rotation in 5 seconds and return
+            conn.mark_last_key_package_to_rotate_in_5s()?;
+            tracing::debug!("Last key package not ready for rotation, scheduled for 5s later");
+            return Ok(());
+        }
 
         let kp = self.new_key_package(provider)?;
         let kp_bytes = kp.tls_serialize_detached()?;
         let hash_ref = serialize_key_package_hash_ref(&kp, provider.crypto())?;
         let history_id = conn.store_key_package_history_entry(hash_ref.clone())?.id;
+        
         match api_client.upload_key_package(kp_bytes, true).await {
             Ok(()) => {
```

Comment:
We might need to clear the `rotate_in` here, or we will keep trying to rotate every time

Reply:
I did it in the Ok state, in case of error I think we should keep the rotate_in value to make sure at the next check the worker catch it and rotate the keys.